data/src/main/java/me/androidbox/data/remote/authentication/AuthenticationImp.kt

```diff
@@ -0,0 +1,29 @@
+package me.androidbox.data.remote.authentication
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import me.androidbox.data.remote.model.request.RegistrationRequestDto
+import me.androidbox.data.remote.network.authentication.AuthenticationService
+import me.androidbox.domain.authentication.remote.Authentication
+import javax.inject.Inject
+
+class AuthenticationImp @Inject constructor(
+    private val authenticationService: AuthenticationService,
+) : Authentication {
+
+    override suspend fun registerUser(fullName: String, email: String, password: String) {
+        CoroutineScope(Dispatchers.IO).launch {
+            /* MOCK DATA */
+            val registrationRequestDto = RegistrationRequestDto(
+                fullName = "JoeBlogs",
+                email = "joe@gmail.com",
+                password = "Password1"
+            )
+
+            authenticationService.register(
+                registrationRequestDto
+            )
+        }
+    }
+}
```

Comment:
Is this the best way to launching a coroutine? By using CoroutineScope

Reply:
I have updated this to use `withContext` so this crash doesn't happen now.
What is the difference to using CoroutineScrope or withContext?

Reply:
CoroutineScope is a way to launch a coroutine
In a viewmodel usually you use viewmodelScope

WithContext, simply changes the context the current code is
So if you are in a Dispatchers.IO (Used for IO Operations like Network and DB), then you can't modify UI. Since that UI needs to be modified in Dispatchers.Main. But, if you run Network or DB on Dispatchers.Main, then the app will freeze since you shouldn't run heavy operations on the Main thread